src/main/java/org/broadinstitute/consent/http/resources/DataAccessRequestResource.java

```diff
@@ -125,21 +125,21 @@ public Response describeManageDataAccessRequests(@Auth AuthUser authUser) {
     @GET
     @Produces("application/json")
     @Path("/manage/v2")
-    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL})
+    @RolesAllowed({ADMIN, CHAIRPERSON, MEMBER, SIGNINGOFFICIAL, RESEARCHER})
     public Response describeManageDataAccessRequestsV2(@Auth AuthUser authUser, @QueryParam("roleName") Optional<String> roleName) {
         try {
             User user = userService.findUserByEmail(authUser.getName());
             String roleNameValue = roleName.orElse(null);
             if (Objects.nonNull(roleNameValue)) {
                 boolean valid = UserRoles.isValidRole(roleNameValue);
                 if (valid) {
-                    //if the roleName is SO and the user does not have that role throw an exception
-                    if (roleNameValue.equalsIgnoreCase(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
-                        if (!user.hasUserRole(UserRoles.SIGNINGOFFICIAL)) {
-                           throw new NotFoundException("User: " + user.getDisplayName() + ", " + " does not have Signing Official role.");
-                       }
-                   } else {
-                        throw new BadRequestException("Signing Official Role is the only role supported at this time");
+                    //if the user does not have the given roleName throw NotFoundException
+                    if (!user.hasUserRole(UserRoles.getUserRoleFromName(roleNameValue))) {
+                        throw new NotFoundException("User: " + user.getDisplayName() + ", does not have " + roleName + " role.");
+                    }
+                    //if there is a valid roleName but it is not SO or Researcher then throw an exception
+                    if (!roleNameValue.equals(UserRoles.RESEARCHER.getRoleName()) && !roleNameValue.equals(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
+                        throw new BadRequestException("Unsupported role name: " + roleName);
                     }
```

Comment:
This is where if-else cases really break down. As more cases are added/removed/handled/etc., this becomes unwieldy. Better to fail fast in a switch block rather than throw iterate over different cases, which is what a switch block is designed to do. Something like this should clean that logic up a lot:

```
                    UserRoles queriedRole = UserRoles.getUserRoleFromName(roleNameValue);
                    switch (queriedRole) {
                        case ADMIN:
                        case CHAIRPERSON:
                        case MEMBER:
                            throw new BadRequestException("Unsupported role name: " + roleNameValue);
                        default:
                            break;
                    }

```
Caveat, there are some more optimizations to be made with the above code, but this is a lot easier to read than the multi-case if statement.